sound/soc/sof/topology.c

```diff
@@ -1437,7 +1437,7 @@ static int sof_widget_load_pcm(struct snd_soc_component *scomp, int index,
 	ret = sof_ipc_tx_message(sdev->ipc, host->comp.hdr.cmd, host,
 				 sizeof(*host), r, sizeof(*r));
 	if (ret >= 0)
-		return ret;
+		return 0;
```

Comment:
can ret be > 0?
If yes, there is an inconsistency with your previous patch.
if no, then the test should be changed as well?

Reply:
@lyakh @plbossart there's no guarantee that the sof_ipc_tx_message() cannot be > 0. It depends on whatever the FW reply contains. But I think the "right" fix would be to change sof_ipc_tx_message() to return 0 if the reply return value is >= 0